tests/data_properties_v3_conversion/genius_zone_data_state_test.py

```diff
@@ -0,0 +1,248 @@
+        "Check that the bOutRequestHeat is correctly set to false"
+
+        self.raw_json["bOutRequestHeat"] = 0
+        genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+        self.assertFalse(genius_zone.data['_state']['bOutRequestHeat'])
+
+    def test_when_bOutRequestHeat_is_true_then_state_bOutRequestHeat_true(self):
+        "Check that the bOutRequestHeat is correctly set to true"
+
+        self.raw_json["bOutRequestHeat"] = 1
+        genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+        self.assertTrue(genius_zone.data['_state']['bOutRequestHeat'])
+
+    def test_when_iMode_set_then_state_mode_is_set_correctly(self):
+        "Check that the mode is set on the class"
+
+        for zone_mode, zone_mode_text in IMODE_TO_MODE.items():
+            with self.subTest(zone_mode=zone_mode, zone_mode_text=zone_mode_text):
+                self.raw_json["iMode"] = zone_mode
+                self.raw_json["zoneSubType"] = 1
+                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+                self.assertEqual(genius_zone.data['mode'], zone_mode_text)
+
+    def test_when_iType_should_set_temperature_state_temperature_set_correctly(self):
+        "Check that the temperature is set for certain values of iType"
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        test_values = [
+            ZONE_TYPE.ControlSP,
+            ZONE_TYPE.TPI,
+            ZONE_TYPE.Manager
+        ]
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
+
+        for zone_type in test_values:
+            with self.subTest(zone_type=zone_type):
+                self.raw_json["iType"] = zone_type
+                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+                self.assertEqual(genius_zone.data['temperature'], temperature)
+
+    def test_when_iType_should_not_set_temperature_state_temperature_not_set(self):
+        "Check that the temperature is not set for certain values of iType"
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        test_values = [
+            ZONE_TYPE.OnOffTimer,
+            ZONE_TYPE.ControlOnOffPID,
+            ZONE_TYPE.Surrogate
+        ]
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
+
+        for zone_type in test_values:
+            with self.subTest(zone_type=zone_type):
+                self.raw_json["iType"] = zone_type
+                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+                self.assertFalse('temperature' in genius_zone.data)
+
+    def test_when_iType_should_set_setpoint_state_setpoint_set_correctly(self):
+        "Check that the setpoint is set for certain values of iType"
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        test_values = [
+            ZONE_TYPE.ControlSP,
+            ZONE_TYPE.TPI
+        ]
```

Comment:
Unless you need a list, use a tuple: `test_values = (ZONE_TYPE.ControlSP, ZONE_TYPE.TPI)`

Reply:
Huh, did not know that! [] sets up an array in C# and Javascript. Cheers.

Reply:
[] is a list, it is mutable (you can add to / delete from / sort it.

() is a tuple, and is like a constant.

Both are iterable (arrays), and the tuple is faster/smaller - more the Python way, in this case.

Reply:
Good to know, thanks!